src/ext/content-scripts/api.js

```diff
@@ -108,8 +108,13 @@ async function setClipboard(clipboardData, type) {
 	});
 }
 
-function xhrResponseProcessor(response) {
-	const res = response;
+/**
+ * Restore `response.response` to required `responseType`
+ * @param {TypeExtMessages.XHRTransportableResponse} msgResponse
+ * @param {TypeExtMessages.XHRResponse} response
+ */
+function xhrResponseProcessor(msgResponse, response) {
+	const res = msgResponse;
```

Comment:
You likely could [destructure](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Destructuring_assignment) this and the `response` arg to avoid the `res.` and `response.` instances. You might need stricter type checks though, so maybe not worth it.

`const { contentType, response: foo, responseType } = res;`